src/mito2/src/read/scan_region.rs

```diff
@@ -891,13 +891,27 @@ impl RangeBuilderList {
         reader_metrics: &mut ReaderMetrics,
     ) -> Result<()> {
         let file_index = index.index - self.mem_builders.len();
-        let mut builder_opt = self.file_builders[file_index].lock().await;
-        match &mut *builder_opt {
+        let builder_opt = self.file_builders[file_index].read().await;
```

Comment:
We can remove the global builder list and use per-partition builder list. So a partition won't wait for other partitions. (This happens on UnorderedScan).

I implemented a prototype in https://github.com/GreptimeTeam/greptimedb/commit/3f9bf48161bcc26ace6787894f68c296a87760d7.